share/completions/mdadm.fish

```diff
@@ -0,0 +1,206 @@
+# fish completions for mdadm, as documented under Debian Stretch (mdadm v3.4)
+# Known lacks :
+# - be more contextual : suggest according to available files, devices, arrays…
+# - especially, suggest values for levels, metadata format… according to the already given parameters
+# - some descriptions are not very useful, even a bit Captain Obvious-like
+# - some options could see their meaning in different contexts explained with different completions with proper tests for context
+# - misc and manage modes could be tested for, and the corresponding options proposed only in the right context
+#
+# -n tests have been simplified when __fish_contains_opt …; or _fish_not_contain_opt …: the options in the first test were removed from the second one,
+#       as their absence has already been checked, because this absence caused evaluation of __fish_not_contain_opt
+
+function __fish_mdadm_metadata_options
+	echo "0	Use original 0.90 format superblock"
```

Comment:
Are these literal tab characters?

I'd prefer it if we used the `\t` escape instead.

It's even possible to use something like

```fish
printf '%s\t%s\n' "0" "Use original 0.90 format superblock" \
                          "0.90" "Use original 0.90 format superblock"
# and so on
```